src/components/LocationSelector/LocationSelector.tsx

```diff
@@ -150,20 +153,24 @@ export const LocationSelector: React.FC<Props> = (props) => {
 
         return findCenter(geocoder, location.place_id)
             .then((center: any) => {
+                const lng = typeof center.lng === 'function' ? center.lng() : center.lng;
+                const lat = typeof center.lat === 'function' ? center.lat() : center.lat;
+                const isLocationSelected = selectedLocations
+                    .map((item) => item.id)
+                    .includes(location.id);
                 const locationToAdd = {
                     ...location,
-                    center: {
-                        lng: typeof center.lng === 'function' ? center.lng() : center.lng,
-                        lat: typeof center.lat === 'function' ? center.lat() : center.lat,
-                    },
+                    center: { lat, lng },
                     radius: hasRadius ? radiusDefaultValue : 0,
                 };
 
-                const isLocationSelected = selectedLocations
-                    .map((item) => item.id)
-                    .includes(location.id);
-
-                if (!isLocationSelected) {
+                if (shouldGetAddressInfo) {
+                    getAddressComponents(geocoder, { lat, lng }).then((addressComponents: any) => {
+                        if (!isLocationSelected) {
+                            onAddLocation({ ...locationToAdd, addressComponents });
+                        }
+                    });
+                } else if (!isLocationSelected) {
                     onAddLocation(locationToAdd);
                 }
```

Comment:
I think we can rewrite that in this way:
```
                if (!isLocationSelected) {
                    if (shouldGetAddressInfo) {
                        getAddressComponents(geocoder, { lat, lng }).then((addressComponents) => {
                            onAddLocation({ ...locationToAdd, addressComponents });
                        });
                    } else {
                        onAddLocation(locationToAdd);
                    }
                }
```

Reply:
@mukiienko nice catch, thanks